src/DotNetty.Transport/Channels/ChannelOutboundBuffer.cs

```diff
@@ -359,6 +361,12 @@ public List<ArraySegment<byte>> GetSharedBufferList()
                             //noinspection ConstantValueVariableUse
                             entry.Count = count = buf.IoBufferCount;
                         }
+                        i += count;
+                        if (i > 1024 && Util.IsLinux)
```

Comment:
Having a platform specific check and some hard coded limit here does not look like a very good idea to me. I think it might make more sense to introduce some channel config option like maxBytesPerGatheringWrite so that the outbound buffer behaves consistently across all platforms. Similar to netty Epoll config
https://github.com/netty/netty/blob/b215794de31f28355e4469fcc04782f55076c80c/transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollChannelConfig.java#L33

Reply:
it fix in .net core sdk 2.0.3, it can be removed. also you can review this code and change it better. now socket send will not send full buffer in linux when retuen sockererror.success. ,it dotnetty need fix.

Reply:
First of all, the outbound buffer cached list already has the limit of 1024
https://github.com/StormHub/DotNetty/blob/dev/src/DotNetty.Transport/Channels/ChannelOutboundBuffer.cs#L704
In original Netty, the buffer array is filled with 1024 slots per batch.
https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java#L395
There are really two limits here
1. Maximum bytes gathered
2. Number of io buffer slots
If you do the same above, there is no need for some platform specific checks.

Reply:
1、you can debug it,i know that, java has check iovmax in socket channle . 
 2、it just decalre list capacity ， when you add item more capacity will auto change in .net

Reply:
i will change my code like java ,thanks.

Reply:
For minimum change, what you have to do is to have a limit passed in similar to what Netty does
https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java#L389
And just pass the limit in from socket channel
https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java#L387

It is much better than some platform specific checks hiding in the outbound buffer.